File: day01/inverse_captcha.py
#!/usr/bin/env python3
from sys import argv

script, filename = argv
with open(filename) as f:
    for line in f:
        line = line.rstrip("\n")
        sum1 = 0
        sum2 = 0
        step2 = int(len(line) / 2)
        for i in range(len(line)):
            if line[i] == line[(i+1) % len(line)]:
                sum1 += int(line[i])
            if line[i] == line[(i+step2) % len(line)]:
                sum2 += int(line[i])

        print("result1", sum1)
        print("result2", sum2)
        

File: day01/inverse_captcha.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature ":5.26";
use experimental 'signatures';

while (<>) {
    chomp;
    my @c = split //;
    my $sum1 = 0;
    my $sum2 = 0;
    my $step2 = @c/2;
    for my $i (0..$#c) {
        $sum1 += $c[$i] if $c[$i] == $c[($i+1) % @c];
        $sum2 += $c[$i] if $c[$i] == $c[($i+$step2) % @c];
    }

    say "result1 is $sum1";
    say "result2 is $sum2";
}
